frontend/src/components/task/task-filters.tsx

```diff
@@ -103,4 +150,16 @@ const TaskFilters = ({
   )
 }
 
-export default injectIntl(withRouter(withStyles(styles)(TaskFilters)))
\ No newline at end of file
+// export default injectIntl(withRouter(withStyles(styles)(TaskFilters)))
+const mapStateToProps = (state) => ({
+  tasks: state.tasks.data, // Assuming you have tasks in your Redux state
+  filteredTasks: state.tasks.filteredData,
+});
+const mapDispatchToProps = {
+  filterTasks,
+};
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(injectIntl(withRouter(withStyles(styles)(TaskFilters))));
+
```

Comment:
This should be moved to the container folder, and the components should remain pure. See other examples in our containers folders. Basically, you have the same component as a container importing the pure component, and you import the container in the component that requires it.